eve/service/svc_note.py

```diff
@@ -0,0 +1,172 @@
+
+    def output(self):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
+
+        # transform null-values to str('N/A')
+        for dic in self.data:
+            for key in dic:
+                if dic[key] is None:
+                    dic[key] = 'N/A'
+
+        # transform all values to str
+        for dic in self.data:
+            for key in dic:
+                dic[key] = str(dic[key])
+
+        # order by priority and due_date
+        self.data = sorted(self.data, key=lambda k: k['Priority'])
+        self.data = sorted(self.data, key=lambda k: k['DueDate'])
+
+        # output data
+        console = Console()
+        table = Table(show_header=True,
+                      box=box.ASCII,
+                      #  show_lines=True,
+                      header_style='bold blue')
+        table.add_column('uid', style='dim')
+        table.add_column('creation date', style='dim')
+        table.add_column('alteration date', style='dim')
+        table.add_column('due date', style='dim')
+        table.add_column('priority', style='dim')
+        table.add_column('note text')
+        for dic in self.data:
+            row = list()
+            for key in dic:
+                row.append(dic[key])
+            table.add_row(str(row[0]), str(row[1]), str(row[2]), str(row[3]), str(row[4]), str(row[5]))
+        console.print(table)
+
+    def add_note(self, note_txt, due_date=None, priority=None):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
+
+        # compose new_note
+        uid = int(len(self.data)+1)
+        self.new_note = dict()
+        self.new_note["UID"] = uid
+        self.new_note["CreationDate"] = str(date.today())
+        self.new_note["AlterationDate"] = None
+        self.new_note["DueDate"] = due_date
+        self.new_note["Priority"] = priority
+        self.new_note["NoteTxt"] = note_txt
+
+        # append new_note to old_notes
+        self.data.append(self.new_note)
+
+        # write to JSON-file
+        with open(self.path, 'w', encoding='utf-8') as f:
+            json.dump(self.data, f, indent=2, ensure_ascii=False)
+
+        print(f'New note with uid {uid} added.')
+
+    def del_note(self, uid):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
+
+        # delete data for given uid
+        self.data = list(filter(lambda i: i["UID"] != int(uid), self.data))
+
+        # assign new uids to data
+        count = 0
+        for dic in self.data:
+            count += 1
+            dic["UID"] = count
+
+        # write to JSON-file
+        with open(self.path, 'w', encoding='utf-8') as f:
+            json.dump(self.data, f, indent=2, ensure_ascii=False)
+
+        print(f'Note with uid {uid} deleted.')
+
+    def alter_note(self, uid, due_date=None, priority=None, note_txt=None):
```

Comment:
You could use the `**kwargs` syntax here to have an arbitrary amount of attribute you can edit on a note.

Reply:
If you're not familiar with `**kwargs` then try this...

```python
def my_func(**kwags):
    print(kwargs)


my_func(param1=1, param2=2, something_else='abc', another_one=False)
```

Reply:
That is cool. I implemented that for the method `edit`.